Add README files for all plugins (#6660)
nextflow-io/nextflow: plugins/nf-azure/README.md

# Microsoft Azure plugin for Nextflow

> This plugin provides support for Azure Blob Storage as a file system and Azure Batch as a compute executor for Nextflow pipelines.

Reviewer: ```suggestion
The Microsoft Azure plugin provides support for Azure Blob Storage as a file system, and Azure Batch as a compute executor for Nextflow pipelines.
```

> ## Get Started

Reviewer: ```suggestion
## Get started
```

To use this plugin, add it to your `nextflow.config`:

```groovy
plugins {
    id 'nf-azure'
}
```

Configure your Azure credentials and services:

```groovy
azure {
    storage {
        accountName = '<YOUR STORAGE ACCOUNT NAME>'
        accountKey = '<YOUR STORAGE ACCOUNT KEY>'
    }

    batch {
        endpoint = 'https://<YOUR BATCH ACCOUNT NAME>.<REGION>.batch.azure.com'
        accountName = '<YOUR BATCH ACCOUNT NAME>'
        accountKey = '<YOUR BATCH ACCOUNT KEY>'
    }
}
```

Set the executor and work directory:

```groovy
process.executor = 'azurebatch'
workDir = 'az://<YOUR CONTAINER>/work'
```

## Examples

### Basic Azure Batch Configuration

```groovy
plugins {
    id 'nf-azure'
}

azure {
    storage {
        accountName = 'mystorageaccount'
        accountKey = System.getenv('AZURE_STORAGE_KEY')
    }

    batch {
        endpoint = 'https://mybatchaccount.westeurope.batch.azure.com'
        accountName = 'mybatchaccount'
        accountKey = System.getenv('AZURE_BATCH_KEY')
        autoPoolMode = true
        deletePoolsOnCompletion = true
    }
}

process.executor = 'azurebatch'
workDir = 'az://mycontainer/work'
```

> ### Using Managed Identity

Reviewer: ```suggestion
### Using managed identity
```
